drivers/net/bnxt/rte_pmd_bnxt.c

```diff
@@ -482,7 +478,7 @@ int rte_pmd_bnxt_mac_addr_add(uint8_t port, struct ether_addr *addr,
 	/* If the VF currently uses a random MAC, update default to this one */
 	if (bp->pf.vf_info[vf_id].random_mac) {
 		if (rte_pmd_bnxt_get_vf_rx_status(port, vf_id) <= 0) {
-			bnxt_hwrm_func_vf_mac(bp, vf_id, (uint8_t *)addr);
+			rc = bnxt_hwrm_func_vf_mac(bp, vf_id, (uint8_t *)addr);
```

Comment:
If we bail out, the default MAC won't be delivered to the VF at all.  By not bailing out, we ensure that even if we are unable to set the default MAC, that MAC will be delivered to the function as requested.

As for setting a filter for the default MAC, it seems that someone has to install one.  Usually the VF driver will do so, and it's not needed, but that's not guaranteed.  At some point, we may be able to save a filter by not installing one for the default MAC, but the correct choice isn't clear yet, so the conservative choice was made.